src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,401 @@
+                    m_architecture = getArchitecture(fields.at(INDEX_ARCHITECTURE));
+                    m_location = getLocation(packageReg);
+                    m_name = getName(fields.at(INDEX_NAME), packageReg);
+                    m_vendor = getVendor(packageReg);
+
+                }
+
+                /*
+                 * If the package location is not in application store database provided by Windows then it is thrown away.
+                 * https://answers.microsoft.com/en-us/windows/forum/all/what-is-cprogram-fileswindows-apps-hidden-folder/783b5a18-c44d-46f7-b638-e98054b7c2a8
+                 */
+                if (m_location.find(STORE_APPLICATION_DATABASE) == std::string::npos)
+                {
+                    m_name.clear();
+                    m_vendor.clear();
+                    m_architecture.clear();
+                    m_version.clear();
+                    m_location.clear();
+                }
+            }
+        }
+
+        /*
+         * @brief Gets packages architecture.
+         *
+         * @param[in] field Field where the architecture type is.
+         *
+         * Type are:
+         *          - x64
+         *          - x86
+         *
+         * @return Return app architecture.
+         */
+        const std::string getArchitecture(const std::string& field)
+        {
+            std::string architecture { UNKNOWN_VALUE };
+
+            if (!field.compare("x64"))
+            {
+                architecture = "x86_64";
+            }
+            else if (!field.compare("x86"))
+            {
+                architecture = "i686";
+            }
+
+            return architecture;
+        }
+
+        /*
+         * Getting location where is the data and configuration application.
+         *
+         * @param[in] registry  Registry object pointing to where the data is.
+         *
+         * @return Return package location.
+         */
+        const std::string getLocation(const Utils::Registry& registry)
+        {
+            std::string location;
+
+            return registry.string("PackageRootFolder", location) ? location : UNKNOWN_VALUE;
+        }
+
+        /*
+         * @brief Get application name.
+         *
+         * @param[in] fullName Full application name.
+         * @param[in] registry Registry object pointing to where the data is located.
+         *
+         * @return Return name application.
+         */
+        const std::string getName(const std::string& fullName, const Utils::Registry& registry)
+        {
+            std::string name;
+
+            /*
+             * Example of  fullName
+             *
+             * Microsoft.Windows.Search
+             * Microsoft.SkypeApp
+             */
+            const auto keyName { Utils::split(fullName, '.').back() }; // Will only use the last vector element.
+            constexpr auto INVALID_NAME_APP { "@{" };
+
+            for (const auto& folder : registry.enumerate())
```

Comment:
registry.enumerate() can throw an exception let's catch it please and return the default value.